### ISSUE TITLE
"microsoft.insight" resource provider get stuck in "registering"

### ISSUE THREAD

**Issue by hamedtayebi** [Azure/azure-cli#26801]
### Describe the bug

Trying to register "microsoft.insight" for my subscription via the portal but it gets stuck in registering and eventually gives this message "'Register' operation check timed out on Resource Provider 'microsoft.insights', please refresh resource providers list to check for registration status".

I've tried reregistering and unregistering it but same behavior. To me seems like a connection issue. It's been 2 days now.

The only thing i found on internet:
https://stackoverflow.com/questions/63451493/azure-resource-provider-stuck-in-registering

### Related command

When I run:
 `az provider register --namespace Microsoft.Insights` 
I get: **"Registering is still on-going. You can monitor using 'az provider show -n Microsoft.Insights' "**
and when I run:
`az provider show --namespace microsoft.insights -o table`
I get:
Namespace          | RegistrationPolicy  |  RegistrationState
microsoft.insights |  RegistrationRequired | Registering

### Errors

"'Register' operation check timed out on Resource Provider 'microsoft.insights', please refresh resource providers list to check for registration status".

### Issue script & Debug output

"'Register' operation check timed out on Resource Provider 'microsoft.insights', please refresh resource providers list to check for registration status".

### Expected behavior

Successful registered resource provider. Similar to all other resources

### Environment Summary

Nothing special

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by hamedtayebi** [comment]
The only thing i found on internet: 
https://stackoverflow.com/questions/63451493/azure-resource-provider-stuck-in-registering

**Comment by hamedtayebi** [comment]
more info:
When I run:
 `az provider register --namespace Microsoft.Insights` 
I get: "Registering is still on-going. You can monitor using 'az provider show -n Microsoft.Insights' "
and when I run:
`az provider show --namespace microsoft.insights -o table`
I get:
Namespace           RegistrationPolicy    RegistrationState
------------------  --------------------  -------------------
microsoft.insights  RegistrationRequired  Registering

**Comment by tdbt1985** [comment]
Do you have any update for this issue? I am having the same issue while running terraform

**Comment by navba-MSFT** [comment]
Adding Service team to look into this.

**Comment by zhoxing-ms** [comment]
@meghanagupta95 Could you please take a look at the provider registering issue? Or could you please help route it to the right person?

**Comment by patelavtx** [comment]
Hi,  
I've recently moved subscription (same tenant) and same issue as described here. That is 'microsoft.insights' provider stuck in 'registering'.  I left this overnight still the same and this morning tried to re-register/refresh, but after an hour still the same status:

```
attila10@DESKTOP-2H1J9PA:~$ az provider list --query "[?registrationState=='Registering']" -o table
Namespace              RegistrationState    RegistrationPolicy
---------------------  -------------------  --------------------
microsoft.insights     Registering          RegistrationRequired
Microsoft.Diagnostics  Registering          RegistrationRequired

```

Q1. Can MS support please advise how to remedy this?

**Comment by dulfer** [comment]
Can confirm.
Tried registering the provider in a subscription in our tenant about 6 hours ago, and it's still in registering state.
Requests that require the provider result in HTTP/400.

Re-ran the script in attempt to fix the problem, without desired effect.
Attempted to unregister the provider, which now got it in a permanent 'Unregistering' state.

Command:
`az provider register --namespace microsoft.insights --accept-terms --wait`

Activity details:
```
"action": "microsoft.insights/register/action",
"operationId": "38c26f19-082c-4cb0-b93c-9717b6b6aa7b",
"submissionTimestamp": "2023-07-03T08:34:48Z"
```

**Comment by tdbt1985** [comment]
After leave it running overnight, I got it registerred

**Comment by patelavtx** [comment]
As above,  shows registered now,  having re-registered via GUI yesterday morning.

**Comment by llohrstraeter** [comment]
This issue should not be closed. I'm trying to register this provider and getting the same issue. It's been stuck for hours in my case. Provider registration usually takes a few minutes at most.

**Comment by Sizzl** [comment]
Can also confirm that this has not been fixed - also have an instance stuck registering microsoft.insights.

**Comment by deniscooper** [comment]
we also have the same issue - trying to log with MS support

**Comment by madams-te** [comment]
Also receiving this issue in Eastus

**Comment by ajmomo** [comment]
I am still facing this issue. Did someone manage to get a workaround for it?

**Comment by szczur** [comment]
I have the same problem but for "Microsoft.OperationalInsights"

**Comment by axelkiwi** [comment]
I am also experiencing the issue since yesterday. I'm having difficulties registering the 'Microsoft.OperationalInsights' provider. The status is stuck at registering. I tried unregistering and re-registering, but the problem persists."

**Comment by mapperCoderZ** [comment]
Same problem `az provider register --namespace  Microsoft.OperationalInsights --verbose --wait`. Nothing happens, registering for a while...

**Comment by Passie1982** [comment]
Same problem here. Gave it minimal 24hours to complete. Still registering. Created a support ticket. See if the Azure Support has a solution. If so, I will update this issue.

**Comment by ambarishsatheesh-rs** [comment]
I have this issue for "Microsoft.OperationalInsights" but when using the azurerm provider for Terraform. Managed to work around it for now by setting `skip_provider_registration = true` for now.

**Comment by sassai** [comment]
#28203 also describes this

**Comment by ajmomo** [comment]
Looks like the issue is now resolved. I was able to complete the registration of the provider Microsoft.OperationalInsights.

**Comment by Passie1982** [comment]
I added a solved response to #28203
